web/src/main/java/org/fao/geonet/monitor/link/LinkMonitorService.java

```diff
@@ -0,0 +1,142 @@
+        super.init(resourceManager, geonetContext, logger, reindexInterval);
+    }
+
+    private ArrayList<String> getAllRecords() {
+        try {
+            return getAllRecordsLucene();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    private ArrayList<String> getAllRecordsLucene() throws Exception {
+        ArrayList<String> records = new ArrayList<String>();
+
+        IndexAndTaxonomy indexAndTaxonomy = getGeonetContext().getSearchmanager().getNewIndexReader(null);
+        try {
+            GeonetworkMultiReader reader = indexAndTaxonomy.indexReader;
+
+            int capacity = (int)(reader.maxDoc() / 0.75) + 1;
+            for (int i = 0; i < reader.maxDoc(); i++) {
+                DocumentStoredFieldVisitor idChangeDateSelector = new DocumentStoredFieldVisitor("_uuid", "_changeDate");
+                reader.document(i, idChangeDateSelector);
+                org.apache.lucene.document.Document doc = idChangeDateSelector.getDocument();
+                String uuid = doc.get("_uuid");
+
+                if (uuid == null) {
+                    Log.error(Geonet.INDEX_ENGINE, "Document with no _uuid field skipped! Document is " + doc);
+                    continue;
+                } else {
+                    getLogger().debug(String.format("Link Monitor Service adding uuid '%s'", uuid));
+                    records.add(uuid);
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            getGeonetContext().getSearchmanager().releaseIndexReader(indexAndTaxonomy);
+        }
+
+        return records;
+    }
+
+    private boolean needReindex() {
+        long now = System.currentTimeMillis() / 1000l;
+        return reindexTimestamp < 0 || now - reindexTimestamp >= getReindexInterval();
+    }
+
+    private void reindex(ArrayList<String> records) {
+        getLogger().info("Link Monitor Service is reindexing...");
+
+        for (final String uuid : records) {
+            if (recordMap.get(uuid) == null) {
+                // New record
+                recordMap.put(uuid, new MetadataRecordInfo(uuid));
+            }
+        }
+
+        for (Map.Entry<String, MetadataRecordInfo> record : recordMap.entrySet()) {
+            if (! records.contains(record.getKey())) {
+                // Record was deleted
+                recordMap.remove(record.getKey());
+            }
+        }
+
+        reindexTimestamp = System.currentTimeMillis() / 1000l;
+    }
+
+    @Override
+    public void check() {
+        if (needReindex())
+            reindex(getAllRecords());
+
+        for (Map.Entry<String, MetadataRecordInfo> record : recordMap.entrySet()) {
+            getLogger().debug(String.format("Checking record '%s'", record.getKey()));
+            record.getValue().check();
+            try {
+                Thread.sleep(betweenChecksIntervalMs);
```

Comment:
I commented on that variable up in the class, I think it'd be good to have it configurable. Otherwise we can potentially hammer services on the same server - which may or may not result in the remote end blocking us...